### DISCUSSION TITLE
Communication between golem apps

### DISCUSSION THREAD

**Discussion by JohnStaples** [ThinkR-open/golem#666] [Q&A]
Firstly, thank you for publishing 'golem'.
I have two golem based apps; a primary complex application and a secondary golem app acting as a subject specific .csv data store, generating .rds files into a 'data' folder.
Two questions please; how do I ensure the 'data' is exported, ( I understand this is auto within a package?), and how/where do I facilitate the import of this data into the primary golem app?
I have added the second package as a dependancy within the primary app. 
Any golem specific guide lines would be very appreciated on how to complete the import, as currently I am failing.
Thank you

**Comment by ColinFay** [comment]
Hey @JohnStaples, 

Yes, data are automatically exported. 

One issue when it comes to importing data from another package is that you can't do `@importFrom packageA dataset`.

Here are two ways to do this : 

1. `data("diamonds", package = "ggplot2")`

For example: 

```r
app_server <- function( input, output, session ) {
  # Your application server logic
  data("diamonds", package = "ggplot2")
  print(diamonds)
}
```

2. `diamonds <- ggplot2::diamonds`

```r
app_server <- function( input, output, session ) {
  # Your application server logic
  diamonds <- ggplot2::diamonds
  print(diamonds)
}
```

Cheers, 
Colin

**Reply by JohnStaples** [reply]
Hi Colin
Many thanks for your reply, it is greatly appreciated.
Cheers John

**Comment by JohnStaples** [comment]
Hi Colin
Many thanks for your reply, it is greatly appreciated
Cheers John
